docs/features/shared-access-key.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "Authentication with shared access keys via ASP.NET Core authentication filters"
+layout: default
+---
+
+## Arcus.WebApi.Security.SharedAccessKeyAuthentication
+
+The `Arcus.WebApi.Security` package provides an mechanism that uses shared access keys to authenticate users.
+This authentication process consists of two parts:
+
+1. Looks for the configured HTTP request header that contains the shared access key
+2. Matches this value with the configured **Key Vault** secret
+
+The package allows two ways to configure this type of authentication mechanmism in an <span>ASP.NET</span> application:
+- [Shared access key attribute](#ArcusWebApiSecuritySharedAccessKeyAuthenticationAttribute) allows local authentication on  `Controller` or method level
+- [Shared access key filter](#ArcusWebApiSecuritySharedAccessKeyAUthenticationFilter) allows global authentication 
+
+## Arcus.WebApi.Security.SharedAccessKeyAuthenticationAttribute
+
+### Introduction
+
+The `SharedAccessKeyAuthenticationAttribute` can be added on both `Controller` and method level in an <span>ASP.NET</span> Core application.
+This attribute will then add authentication to the route(s) via shared access keys configurable on the attribute itself.
+
+### Usage
+
+The authentication requires a `ICachedSecretProvider` or `ISecretProvider` to be registered in services of the applications (normally in the `Startup` class):
+
+```csharp
+public void ConfigureServices(IServiceCollections services)
+{
+    services.AddScoped<ICachedSecretProvider>(serviceProvider => new CachedSecretProvider(new MySecretProvider()));
+    services.AddMvc();
+}
+```
+
+After that, you can freely use the attribute on the route (single method) or routes (multiple methods or `Controller` level) that requires authentication:
+
+```csharp
+[ApiController]
+[SharedAccessKeyAuthentication(headerName: "http-request-header-name", secretName: "shared-access-key-name")]
+public class MyApiController : ControllerBase
+{
+    [HttpGet]
+    [Route("authz/shared-access-key")]
+    public Task<IActionResult> AuthorizedGet()
+    {
+        return Task.FromResult<IActionResult>(Ok());
+    }
+}
+```
+
+## Arcus.WebApi.Security.SharedAccessKeyAuthenticationFilter
```

Comment:
Maybe it would make sense to change this to `Enforce shared access key globally` and move this before our other approach to stimulate this one.

Reply:
Don't forget to update the link above

Reply:
I don't follow, what do you want to update? I changed the short description with `Enforce...`.

Reply:
Where did you update this?

When I look at the latest version I still see this, but might be doing something wrong then 🤔 

![image](https://user-images.githubusercontent.com/4345663/57446237-8e326f00-7254-11e9-860f-101e6dd22a3a.png)

Reply:
Could you rephrase what you want to change from the original?

Reply:
The screenshot is a good example - The line above my comment (the ##) should be something along my suggestion